spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -798,14 +798,14 @@ private static class Context implements Serializable {
       this.dataFileFormat = writeConf.dataFileFormat();
       this.targetDataFileSize = writeConf.targetDataFileSize();
       this.deleteSparkType = info.rowIdSchema().get();
-      this.deleteFileFormat = writeConf.deleteFileFormat();
+      this.useDVs = writeConf.useDVs();
+      this.deleteFileFormat = useDVs ? FileFormat.PUFFIN : writeConf.deleteFileFormat();
```

Comment:
actually, we should probably just get this from the conf (embed the logic to determine if DVs are used there)